packages/react-widgets/src/widgets/FormulaWidget.js

```diff
@@ -31,28 +32,35 @@ function FormulaWidget({
   joinOperation,
   formatter,
   animation,
+  global,
   onError,
   wrapperProps
 }) {
   const isSourceReady = useSelector((state) =>
     selectAreFeaturesReadyForSource(state, dataSource)
   );
-  const { filters, filtersLogicalOperator } = useSourceFilters({ dataSource, id });
+  const source = useWidgetSource({ dataSource, id });
 
   const [formulaData, setFormulaData] = useState(null);
   const [isLoading, setIsLoading] = useState(true);
 
+  const spatialFilter = useSelector((state) =>
+    global ? selectSpatialFilter(state, source?.id) : null
+  );
+
+  const isReady = global ? true : isSourceReady;
```

Comment:
Maybe you can keep the same pattern as the selector above and put the ternary on `global` in the `isSourceReady`.
That way we don't need that many vars to track the readyness.

Reply:
I like it hehe. Thanks.